sdk/helper/keysutil/policy_test.go

```diff
@@ -989,3 +991,137 @@ func isUnsupportedGoHashType(hashType HashType, err error) bool {
+			wantStr: "aes128-gcm96",
+		},
+		{
+			kt:      KeyType_AES256_GCM96,
+			wantEnc: true, wantDec: true,
+			wantSign:    false,
+			wantHashSig: false, wantDeriv: true,
+			wantKeyAgree: false, wantAssoc: true, wantImp: false,
+			wantStr: "aes256-gcm96",
+		},
+		{
+			kt:      KeyType_ChaCha20_Poly1305,
+			wantEnc: true, wantDec: true,
+			wantSign:    false,
+			wantHashSig: false, wantDeriv: true,
+			wantKeyAgree: false, wantAssoc: true, wantImp: false,
+			wantStr: "chacha20-poly1305",
+		},
+		{
+			kt:      KeyType_XChaCha20_Poly1305,
+			wantEnc: true, wantDec: true,
+			wantSign:    false,
+			wantHashSig: false, wantDeriv: true,
+			wantKeyAgree: false, wantAssoc: true, wantImp: false,
+			wantStr: "xchacha20-poly1305",
+		},
+		// --- ECDSA / ED25519 ---
+		{
+			kt:      KeyType_ECDSA_P256,
+			wantEnc: false, wantDec: false,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: true, wantAssoc: false, wantImp: true,
+			wantStr: "ecdsa-p256",
+		},
+		{
+			kt:      KeyType_ECDSA_P384,
+			wantEnc: false, wantDec: false,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: true, wantAssoc: false, wantImp: true,
+			wantStr: "ecdsa-p384",
+		},
+		{
+			kt:      KeyType_ECDSA_P521,
+			wantEnc: false, wantDec: false,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: true, wantAssoc: false, wantImp: true,
+			wantStr: "ecdsa-p521",
+		},
+		{
+			kt:      KeyType_ED25519,
+			wantEnc: false, wantDec: false,
+			wantSign:    true,
+			wantHashSig: false, wantDeriv: true,
+			wantKeyAgree: false, wantAssoc: false, wantImp: true,
+			wantStr: "ed25519",
+		},
+		// --- RSA ---
+		{
+			kt:      KeyType_RSA2048,
+			wantEnc: true, wantDec: true,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: false, wantAssoc: false, wantImp: true,
+			wantStr: "rsa-2048",
+		},
+		{
+			kt:      KeyType_RSA3072,
+			wantEnc: true, wantDec: true,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: false, wantAssoc: false, wantImp: true,
+			wantStr: "rsa-3072",
+		},
+		{
+			kt:      KeyType_RSA4096,
+			wantEnc: true, wantDec: true,
+			wantSign:    true,
+			wantHashSig: true, wantDeriv: false,
+			wantKeyAgree: false, wantAssoc: false, wantImp: true,
+			wantStr: "rsa-4096",
+		},
+		// --- HMAC ---
+		{
+			kt:      KeyType_HMAC,
+			wantEnc: false, wantDec: false,
+			wantSign:    false,
+			wantHashSig: false, wantDeriv: false,
+			wantKeyAgree: false, wantAssoc: false, wantImp: false,
+			wantStr: "hmac",
+		},
+		// -- Unknown ---
+		{
+			kt:      KeyType(999),
+			wantEnc: false, wantDec: false,
+			wantSign:    false,
+			wantHashSig: false, wantDeriv: false,
+			wantKeyAgree: false, wantAssoc: false, wantImp: false,
+			wantStr: "[unknown]",
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.kt.String(), func(t *testing.T) {
+			assert.Equal(t, tc.wantEnc, tc.kt.EncryptionSupported())
```

Comment:
I'm not against using assert but we mostly use require from the same package fwiw.

Reply:
thanks for noticing fixed.